ckanext/unaids/theme/templates/scheming/form_snippets/upload.html

```diff
@@ -37,8 +37,15 @@ <h3 class="text-muted">
     {% else %}
         <div class="dropzone text-muted">
             <h3><i class="fa fa-spinner fa-spin"></i> {{ _('Loading') }}</h3>
-        </div>        
+        </div>
     {% endif %}
+
+    <input name="url" data-testid="url" type="hidden" value="{{ data.url if data else '' }}">
+    <input name="url_type" data-testid="url_type" type="hidden" value="{{ data.url_type if data else '' }}">
+    <input name="lfs_prefix" data-testid="lfs_prefix" type="hidden" value="{{ data.lfs_prefix if data else '' }}">
+    <input name="sha256" data-testid="sha256" type="hidden" value="{{ data.sha256 if data else '' }}">
+    <input name="size" data-testid="size" type="hidden" value="{{ data.size if data else '' }}">
```

Comment:
if we always send empty `size`, `sha256` and `lfs_prefix` will not allow users to create empty resources (without resource fiels).

Changing it to:
```
{% if data.lfs_prefix %}
<input name="lfs_prefix" data-testid="lfs_prefix" type="hidden" value="{{ data.lfs_prefix}}">
{% endif %}
```
only partially resolves the problem as it'll still not allow users to **remove** the file from a resource in an already existing resource.
I don't know how to enable support for removing the resource files.

Reply:
We have spotted that this is actually a problem with the validator logic.  It should be possible for giftless metadata to be all empty strings. I have updated the PR to include the fix to the validator logic.